scrapy/extensions/feedexport.py

```diff
@@ -367,16 +369,18 @@ def _exporter_supported(self, format):
             return True
         logger.error("Unknown feed format: %(format)s", {'format': format})
 
-    def _settings_are_valid(self, uri):
+    def _settings_are_valid(self):
         """
-        If FEED_STORAGE_BATCH_ITEM_COUNT setting or FEEDS.batch_item_count is specified uri has to contain
+        If FEED_EXPORT_BATCH_ITEM_COUNT setting or FEEDS.batch_item_count is specified uri has to contain
         %(batch_time)s or %(batch_id)s to distinguish different files of partial output
         """
         for uri_template, values in self.feeds.items():
-            if values['batch_item_count'] and not any(s in uri_template for s in ['%(batch_time)s', '%(batch_id)s']):
+            if values['batch_item_count'] and not re.findall(r'(%\(batch_time\)s|(%\(batch_id\)0\d*d))', uri_template):
```

Comment:
What about just looking for `%(asdf)`, and hope that the user did not break the formatting syntax? Either we do that, or we implement something that finds all [syntax variants](https://docs.python.org/3/library/stdtypes.html#old-string-formatting).

Also, `findall` probably does not make sense here, given the preceding `not`; finding a single match should suffice in this scenario.

Reply:
> What about just looking for %(asdf), and hope that the user did not break the formatting syntax? Either we do that, or we implement something that finds all syntax variants.

@Gallaecio, could you please explain it a little bit more because I can't clearly understand why should we allow users to break the formatting syntax if batch_id is always an integer and we only want to add leading zeros?

Reply:
My thinking was that we should allow as much flexibility as possible if it’s trivial to do. For example, if someone wants to use spaces instead of zeroes for padding, we would be allowing that (`%(batch_id)4d`).

Not a strong opinion about it, though.